autogen/main/firewall.tf.tmpl

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+{{ autogeneration_note }}
+
+
+/******************************************
+  Match the gke-<CLUSTER>-<ID>-all INGRESS
+  firewall rule created by GKE but for EGRESS
+
+  Required for clusters when VPCs enforce
+  a default-deny egress rule
+ *****************************************/
+resource "google_compute_firewall" "intra_egress" {
+  count       = var.firewall_enabled ? 1 : 0
+  name        = "gke-${substr(var.name, 0, min(25, length(var.name)))}-intra-cluster-egress"
```

Comment:
Thanks for the contribution, looks like tests are failing due to duplicate fw names being given to two fixtures `beta-cluster-local` and `sandbox-enabled-local`. Both use the same example [simple-regional-beta](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/examples/simple_regional_beta/main.tf) with a unique prefix. With the substring in the FW name, it takes off that prefix and hence the clash. 

One possibility is to provide a default value for cluster name in the example and override them in the fixtures to set names as beta-cluster and sandbox-enabled to get around this?

Reply:
Maybe we could switch from suffix to prefix?

Reply:
Do we want firewall support enabled on all examples/tests or should we create a dedicated example for this instead? Because this is going to be disabled by default now so these breaking tests will presumably pass again.

Reply:
Let's enable it one one of the examples, maybe `shared_vpc`. We can leave the default on the rest.

Reply:
Enabled on `shared_vpc`

Reply:
Could either of you explain why https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/runs/605538699 failed? I did just run the `shared-vpc-local` test locally and it passed.

Reply:
@Dev25 looks like it was some thing unrelated and worked when I reran it.